File: Algorithms/Sorting/Selection Sort/selection_sort.py
arr = [9, 3, 8, 5, 6, 7, 2, 1, 4]


def selection_sort(arr):
    for i in range(len(arr)):
        minIndex = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[minIndex]:
                minIndex = j
        [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]]


selection_sort(arr)
print(arr)

File: Algorithms/Sorting/Selection Sort/selectionSort.js
let arr = [1, 44, 5, 78, 22, 4, 9, 90];

function sort(arr) {
  // Dividing the whole unsorted array into two parts (sorted/unsorted).
  for (let i = 0; i < arr.length; i++) {
    // Getting Indicator to The Current Element.
    let minIndex = i;
    for (let j = i + 1; j < arr.length; j++) {
      // Now, We are comparing and reAssign the minIndex as it'll be the smallest value.
      if (arr[minIndex] > arr[j]) {
        minIndex = j;
      }
    }
    // now we swap the smallest for the current iterator.
    [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
  }
}

sort(arr);
console.log(arr);
